### PR TITLE
Get rid of ApparentBubble

### PR DESCRIPTION
**Pull request by eggrobin**
Add an `ApparentWorld` for consistency: uncorrected coordinates come in as `ApparentWorld`, are converted to `Apparent`, then to `ApparentPileUp` via a `MechanicalSystem` in the `PileUp`.

The correction produces `NonRotatingPileUp` coordinates, which eventually make it back to the game in `World`.

### CODE REVIEW

File: ksp_plugin/interface.hpp [hunk 1]
```diff
@@ -43,6 +43,7 @@ using geometry::R3Element;
 using geometry::Quaternion;
 using geometry::Vector;
 using geometry::Velocity;
+using ksp_plugin::ApparentWorld;
```
Comment (pleroy) [inline ksp_plugin/interface.hpp:46]: Albaphetize.

File: ksp_plugin/pile_up.cpp [hunk 2]
```diff
@@ -325,8 +325,8 @@ not_null<std::unique_ptr<PileUp>> PileUp::ReadFromMessage(
          message.apparent_part_degrees_of_freedom()) {
       pile_up->apparent_part_rigid_motion_.emplace(
           part_id_to_part(part_id),
-          RigidMotion<RigidPart, ApparentBubble>::MakeNonRotatingMotion(
-              DegreesOfFreedom<ApparentBubble>::ReadFromMessage(
+          RigidMotion<RigidPart, Apparent>::MakeNonRotatingMotion(
+              DegreesOfFreedom<Apparent>::ReadFromMessage(
                   degrees_of_freedom)));
```
Comment (pleroy) [inline ksp_plugin/pile_up.cpp:330]: Might fit on the previous line.

File: ksp_plugin/pile_up.cpp [hunk 3]
```diff
@@ -341,7 +341,7 @@ not_null<std::unique_ptr<PileUp>> PileUp::ReadFromMessage(
          message.apparent_part_rigid_motion()) {
       pile_up->apparent_part_rigid_motion_.emplace(
           part_id_to_part(part_id),
-          RigidMotion<RigidPart, ApparentBubble>::ReadFromMessage(
+          RigidMotion<RigidPart, Apparent>::ReadFromMessage(
               rigid_motion));
```
Comment (pleroy) [inline ksp_plugin/pile_up.cpp:345]: Same.

File: ksp_plugin/plugin.cpp [hunk 4]
```diff
@@ -656,32 +656,33 @@ void Plugin::FreeVesselsAndPartsAndCollectPileUps(Time const& Δt) {
...
-      World::unmoving};
+  RigidMotion<ApparentWorld, Apparent> world_to_apparent{
+      RigidTransformation<ApparentWorld, Apparent>{
+          ApparentWorld::origin,
+          Apparent::origin,
+          OrthogonalMap<Barycentric, Apparent>::Identity() *
+              renderer_->WorldToBarycentric(PlanetariumRotation()) *
+              OrthogonalMap<World, ApparentWorld>::Identity()},
+      Identity<World, ApparentWorld>()(renderer_->BarycentricToWorld(
+          PlanetariumRotation())(-angular_velocity_of_world_)),
```
Comment (pleroy) [inline ksp_plugin/plugin.cpp:677]: This `-` sign is frightening.  I wonder if we could hide it in a linear transformation (what it probably does is to convert the angular velocity of A wrt B into the angular velocity of B wrt A).

File: ksp_plugin_test/pile_up_test.cpp [hunk 5]
```diff
@@ -161,52 +161,52 @@ class PileUpTest : public testing::Test {
...
-    // Centre of mass of |p1_| and |p2_| in |ApparentBubble|, in SI units:
+    // Centre of mass of |p1_| and |p2_| in |Apparent|, in SI units:
     //   {1 / 9, -1 / 3, -2 / 9} {10 / 9, -10 / 3, -20 / 9}
-    DegreesOfFreedom<ApparentBubble> const p1_dof(
-        ApparentBubble::origin +
-            Displacement<ApparentBubble>({-11.0 / 3.0 * Metre,
+    DegreesOfFreedom<Apparent> const p1_dof(
+        Apparent::origin +
+            Displacement<Apparent>({-11.0 / 3.0 * Metre,
                                           -1.0 * Metre,
```
Comment (pleroy) [inline ksp_plugin_test/pile_up_test.cpp:169]: Please fix the indentation.

File: physics/massive_body_body.hpp [hunk 6]
```diff
@@ -156,7 +156,8 @@ inline not_null<std::unique_ptr<MassiveBody>> MassiveBody::ReadFromMessage(
         switch (static_cast<Tag>(enum_value_descriptor->number())) {
           ROTATING_BODY_TAG_VALUE_CASE(ALICE_SUN);
           ROTATING_BODY_TAG_VALUE_CASE(ALICE_WORLD);
-          ROTATING_BODY_TAG_VALUE_CASE(APPARENT_BUBBLE);
+          ROTATING_BODY_TAG_VALUE_CASE(APPARENT);
+          ROTATING_BODY_TAG_VALUE_CASE(APPARENT_WORLD);
```
Comment (pleroy) [inline physics/massive_body_body.hpp:160]: Also add `PILE_UP_PRINCIPAL_AXES`, it's missing in this statement.

File: ksp_plugin/plugin.cpp (hunk 4 above)
Comment (eggrobin) [reply ksp_plugin/plugin.cpp:677]: Yeah this is the angular velocity *of* world, but we are giving the angular velocity *in* world (of apparent which rotates like barycentric).

I think now we have more constructors for RigidTransformation, maybe we can use the angular velocity of world more directly.

File: physics/massive_body_body.hpp (hunk 6 above)
Comment (eggrobin) [reply physics/massive_body_body.hpp:160]: Done, but we should probably restrict this if not to the frames we actually want to use for bodies, at least to inertial ones which is a requirement of `RotatingBody`…

File: ksp_plugin_test/pile_up_test.cpp [hunk 7]
```diff
@@ -165,12 +165,11 @@ class PileUpTest : public testing::Test {
...
+            Displacement<Apparent>(
+                {-11.0 / 3.0 * Metre, -1.0 * Metre, 2.0 / 3.0 * Metre}),
         Velocity<Apparent>({-110.0 / 3.0 * Metre / Second,
-                                  -10.0 * Metre / Second,
-                                  20.0 / 3.0 * Metre / Second}));
+                            -10.0 * Metre / Second,
+                            20.0 / 3.0 * Metre / Second}));
     DegreesOfFreedom<Apparent> const p2_dof(
         Apparent::origin +
             Displacement<Apparent>({2.0 * Metre,
```
Comment (pleroy) [inline ksp_plugin_test/pile_up_test.cpp:175]: More bad ident below this line; please check the entire file.

File: physics/rigid_motion.hpp [hunk 8]
```diff
@@ -65,7 +65,8 @@ class RigidMotion final {
 
   template<typename F = FromFrame,
            typename T = ToFrame,
-           typename = std::enable_if_t<std::is_same_v<F, T>>>
+           typename = typename = std::enable_if_t<
+               F::handedness == T::handedness && F::motion <= T::motion>>
```
Comment (pleroy) [inline physics/rigid_motion.hpp:69]: This needs a comment.

**Pull request merged**